### PR TITLE
Fill in missing filename after broadcast within same question tile

### PR DESCRIPTION
**Pull request by minhaminha**
## Product Description

In the very specific scenario in which there's a multimedia question and a text question that receives the broadcasted filename of the upload AND they are both placed in the same question tile, the filename will disappear from the multimedia question and then claim there is not file there at all. This fixes that to work normally.

Due to some of the workarounds used the UI is now also slightly different:
Before:
Screen Shot 2024-08-26 at 1 59 17 PM

After:
Screen Shot 2024-08-26 at 1 58 10 PM

The difference is much less noticeable in webapps. 

## Technical Summary

[Jira Ticket](https://dimagi.atlassian.net/browse/USH-4690)

This bug gave me a lot of trouble but it boils down to 2 realizations:
1. When a hidden question's display condition is met and needs to become visible, the whole question container needs to be regenerated so that it can now display the question. When the multimedia question is in the same container, this has the unintended effect of re-creating another multimedia question, this time unanswered and un-populated by a file upload. This is why the filename disappears and if the question is required, will raise an error about the question being blank. 
2. You can not edit the text of an input element. The "No file selected" text that's replaced by the filename is handled only by the element and no amount of JS trickery can change that. 

So I went another route and made a pseudo input element by having a button that points to the actual, hidden upload element and a fully controllable in-line text element. 

### Safety story

Basically only changes UI but may have unknown consequences because this is a niche function (will go through QA). Doesn't interact with any data (the filename that it ends up displaying is not the internal, unique filename used to keep track of the media. 

### QA Plan

[QA ticket](https://dimagi.atlassian.net/browse/QA-7039)

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js [hunk 1]
```diff
@@ -931,26 +931,35 @@ hqDefine("cloudcare/js/form_entry/entries", [
         // Tracks whether file entry has already been cleared, preventing an additional failing request to Formplayer
         self.cleared = false;
         self.buildBroadcastTopics(options);
+        self.uploadFile = function() {
+            document.getElementById(self.entryId).click();
+        }
+        self.fileNameDisplay = ko.observable(constants.NO_FILE_SELECTED)
```
Comment (MartinRiese) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js]: translation here?

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js [hunk 2]
```diff
@@ -1004,6 +1013,7 @@ hqDefine("cloudcare/js/form_entry/entries", [
         self.question.error(null);
         self.question.onClear();
         self.broadcastMessages(self.question, constants.NO_ANSWER);
+        self.fileNameDisplay(constants.NO_FILE_SELECTED);
```
Comment (MartinRiese) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js:1016]: and translation here?

File: corehq/apps/cloudcare/templates/cloudcare/partials/form_entry/entry_file.html [hunk 3]
```diff
@@ -2,15 +2,20 @@
 
 <script type="text/html" id="file-entry-ko-template">
   <div class="d-flex">
-    <input type="file" data-bind="
+    <input type="file" style="display: none;" data-bind="
```
Comment (MartinRiese) [inline corehq/apps/cloudcare/templates/cloudcare/partials/form_entry/entry_file.html:5]: we try to not use inline style. You could use `class="d-none"` if bootstrap is available here.

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js [hunk 4]
```diff
@@ -690,6 +690,14 @@ hqDefine("cloudcare/js/form_entry/form_ui", [
                 if (element.serverError) {
                     element.serverError(null);
                 }
+
+                if (allChildren) {
+                    for (let i = 0; i < allChildren.length; i++) {
+                        if (allChildren[i].control >= constants.CONTROL_IMAGE_CHOOSE) {
```
Comment (MartinRiese) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js:696]: not sure I understand what this does?

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js (hunk 1 above)
Comment (minhaminha) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js]: I'm not super familiar with how to enable strings in JS to be translatable - is there a JS equivalent to Python's `gettext` that we frequently use?

Comment (Robert-Costello) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js]: You can use `gettext` https://commcare-hq.readthedocs.io/translations.html#tagging-strings-in-javascript

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js [hunk 5]
```diff
@@ -934,7 +934,7 @@ hqDefine("cloudcare/js/form_entry/entries", [
         self.uploadFile = function () {
             document.getElementById(self.entryId).click();
         };
-        self.fileNameDisplay = ko.observable(constants.NO_FILE_SELECTED);
+        self.fileNameDisplay = ko.observable(gettext("No file selected."));
```
Comment (MartinRiese) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js:937]: I think you could have kept the contant

**Review by MartinRiese — APPROVED** [review]

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js (hunk 5 above)
Comment (minhaminha) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js:937]: I figured this particular string isn't going to be used anywhere else in the future so I'd just hard define them where they're needed and not an unnecessary line to the const.js file
